app/src/main/java/com/eveningoutpost/dexdrip/NSEmulatorReceiver.java

```diff
@@ -247,83 +248,116 @@ private JSONObject extractParams(final Bundle bundle) {
-        if(json_object == null) {
+        if (json_object == null) {
             return;
         }
         String decoded_buffer;
         String patchUidString;
         String patchInfoString;
         String tagId;
         long CaptureDateTime;
+        int[] trend_bg_vals = null;
+        int[] history_bg_vals = null;
+
         try {
             decoded_buffer = json_object.getString(Intents.DECODED_BUFFER);
             patchUidString = json_object.getString(Intents.PATCH_UID);
             patchInfoString = json_object.getString(Intents.PATCH_INFO);
             tagId = json_object.getString(Intents.TAG_ID);
             CaptureDateTime = json_object.getLong(Intents.LIBRE_DATA_TIMESTAMP);
+            if (json_object.has(Intents.TREND_BG) && json_object.has(Intents.HISTORIC_BG)) {
+                JSONArray computed_bg = json_object.getJSONArray(Intents.TREND_BG);
+                trend_bg_vals = new int[computed_bg.length()];
+                for (int i = 0; i < computed_bg.length(); i++) {
+                    trend_bg_vals[i] = computed_bg.getInt(i);
+                }
+                computed_bg = json_object.getJSONArray(Intents.HISTORIC_BG);
+                history_bg_vals = new int[computed_bg.length()];
+                for (int i = 0; i < computed_bg.length(); i++) {
+                    history_bg_vals[i] = computed_bg.getInt(i);
+                }
+            }
         } catch (JSONException e) {
             Log.e(TAG, "Error JSONException ", e);
             return;
         }
-        if(decoded_buffer == null) {
+        if (decoded_buffer == null) {
             Log.e(TAG, "Error could not get decoded_buffer");
             return;
         }
-        
+
         // Does this throws exception???
         byte[] fram_data = Base64.decode(decoded_buffer, Base64.NO_WRAP);
-        byte []patchUid = Base64.decode(patchUidString, Base64.NO_WRAP);
-        byte []patchInfo = Base64.decode(patchInfoString, Base64.NO_WRAP);
-        LibreOOPAlgorithm.handleOop2DecodeFramResult(tagId, CaptureDateTime, fram_data, patchUid, patchInfo);
+        byte[] patchUid = Base64.decode(patchUidString, Base64.NO_WRAP);
+        byte[] patchInfo = Base64.decode(patchInfoString, Base64.NO_WRAP);
+        LibreOOPAlgorithm.handleOop2DecodeFramResult(tagId, CaptureDateTime, fram_data, patchUid, patchInfo, trend_bg_vals, history_bg_vals);
     }
-    
+
     private void handleOop2DecodeBleResult(Bundle bundle) {
-        if(Pref.getBooleanDefaultFalse("external_blukon_algorithm")) {
+        if (Pref.getBooleanDefaultFalse("external_blukon_algorithm")) {
             Log.e(TAG, "External OOP algorithm is on, ignoring ble decrypted data.");
             return;
         }
         JSONObject json_object = extractParams(bundle);
-        if(json_object == null) {
+        if (json_object == null) {
             return;
         }
         String decoded_buffer;
         String patchUidString;
+        JSONArray computed_bg;
```

Comment:
No need to declare it here. It can be declared at line 327. In fact, it is identical to the corresponding part in `handleOop2DecodeFramResult`.